bodo/pandas/physical/read_parquet.h

```diff
@@ -1,10 +1,93 @@
 #pragma once
 
+#include <Python.h>
+#include <arrow/compute/api.h>
+#include <arrow/python/pyarrow.h>
 #include <memory>
 #include <utility>
+#include "../io/arrow_compat.h"
 #include "../io/parquet_reader.h"
+#include "_duckdb_util.h"
 #include "operator.h"
 
+std::function<arrow::compute::Expression(arrow::compute::Expression,
+                                         arrow::compute::Expression)>
+expressionTypeToArrowCompute(const duckdb::ExpressionType &expr_type) {
+    switch (expr_type) {
+        case duckdb::ExpressionType::COMPARE_EQUAL:
+            return arrow::compute::equal;
+        case duckdb::ExpressionType::COMPARE_NOTEQUAL:
+            return arrow::compute::not_equal;
+        case duckdb::ExpressionType::COMPARE_GREATERTHAN:
+            return arrow::compute::greater;
+        case duckdb::ExpressionType::COMPARE_LESSTHAN:
+            return arrow::compute::less;
+        case duckdb::ExpressionType::COMPARE_GREATERTHANOREQUALTO:
+            return arrow::compute::greater_equal;
+        case duckdb::ExpressionType::COMPARE_LESSTHANOREQUALTO:
+            return arrow::compute::less_equal;
+        default:
+            throw std::runtime_error("Unhandled comparison expression type.");
+    }
+}
+
+PyObject *tableFilterSetToArrowCompute(duckdb::TableFilterSet &filters,
+                                       PyObject *schema_fields) {
+    PyObject *ret = Py_None;
+    if (filters.filters.size() == 0) {
+        return ret;
+    }
+    arrow::py::import_pyarrow_wrappers();
+    std::vector<PyObject *> to_be_freed;
+
+    if (filters.filters.size() > 1) {
+        throw std::runtime_error(
+            "tableFilterSetToPhysicalExpression currently supports only a "
+            "single filter.");
+    }
+
+    for (auto &tf : filters.filters) {
+        switch (tf.second->filter_type) {
+            case duckdb::TableFilterType::CONSTANT_COMPARISON: {
+                duckdb::unique_ptr<duckdb::ConstantFilter> constantFilter =
+                    dynamic_cast_unique_ptr<duckdb::ConstantFilter>(
+                        std::move(tf.second));
+                PyObject *py_selected_field =
+                    PyList_GetItem(schema_fields, tf.first);
```

Comment:
`PyList_GetItem` returns a borrowed reference so should not `Py_DECREF`.

https://docs.python.org/3/c-api/list.html#c.PyList_GetItem